tests/python_tests/test_whisper_pipeline.py

```diff
@@ -166,13 +179,60 @@ def get_samples_from_dataset(
     return [x["audio"]["array"] for x in ds]
 
 
-@pytest.mark.parametrize("model_descr", get_whisper_models_list())
-@pytest.mark.parametrize("dataset_id", ["hf-internal-testing/librispeech_asr_dummy"])
-@pytest.mark.precommit
-def test_whisper_on_hf_dataset(model_descr, dataset_id):
-    model_id, path, opt_pipe, genai_pipe = read_whisper_model(model_descr)
+def run_pipeline_with_ref(
+    model_id: str,
+    tmp_path: str,
+    sample: np.ndarray | list[np.ndarray],
+    generation_config: GenerationConfig | None = None,
+    print_infer_time=False,
+):
+    _, _, hf_pipe, genai_pipe = read_whisper_model((model_id, tmp_path))
+
+    if type(sample) is np.ndarray and len(sample.shape) == 1:
+        sample = np.expand_dims(sample, 0)
+
+    hf_infer_time, genai_infer_time = 0, 0
+    hf_result, genai_result = None, None
+    for _sample in sample:
+        start = time.time()
+        genai_result = run_genai(genai_pipe, _sample, generation_config)
+        genai_infer_time += time.time() - start
+
+        start = time.time()
+        hf_result = run_huggingface(hf_pipe, _sample, generation_config)
+        hf_infer_time += time.time() - start
+
+        compare_results(hf_result, genai_result)
+
+    if print_infer_time:
+        print(f"\nInference time HF: {hf_infer_time:.2f} GenAI: {genai_infer_time:.2f}")
```

Comment:
What is it for? Infer times aren't asserted, they are just printed.

If it's going to stay, the time should be measured with time.perf_counter()

Reply:
It was for debugging, removed